src/renderers/dom/shared/__tests__/ReactDOMAttribute-test.js

```diff
@@ -0,0 +1,116 @@
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+describe('ReactDOM unknown attribute', () => {
+  var React;
+  var ReactDOM;
+
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+    ReactDOM = require('react-dom');
+  });
+
+  function normalizeCodeLocInfo(str) {
+    return str && str.replace(/\(at .+?:\d+\)/g, '(at **)');
+  }
+
+  function testUnknownAttributeRemoval(givenValue) {
+    var el = document.createElement('div');
+    ReactDOM.render(<div unknown="something" />, el);
+    expect(el.firstChild.getAttribute('unknown')).toBe('something');
+    ReactDOM.render(<div unknown={givenValue} />, el);
+    expect(el.firstChild.hasAttribute('unknown')).toBe(false);
+  }
+
+  function testUnknownAttributeAssignment(givenValue, expectedDOMValue) {
+    var el = document.createElement('div');
+    ReactDOM.render(<div unknown="something" />, el);
+    expect(el.firstChild.getAttribute('unknown')).toBe('something');
+    ReactDOM.render(<div unknown={givenValue} />, el);
+    expect(el.firstChild.getAttribute('unknown')).toBe(expectedDOMValue);
+  }
+
+  describe('unknown attributes', () => {
+    it('removes values null and undefined', () => {
+      testUnknownAttributeRemoval(null);
+      testUnknownAttributeRemoval(undefined);
+    });
+
+    it('removes unknown attributes that were rendered but are now missing', () => {
+      var el = document.createElement('div');
+      ReactDOM.render(<div unknown="something" />, el);
+      expect(el.firstChild.getAttribute('unknown')).toBe('something');
+      ReactDOM.render(<div />, el);
+      expect(el.firstChild.hasAttribute('unknown')).toBe(false);
+    });
+
+    it('passes through strings', () => {
+      testUnknownAttributeAssignment('a string', 'a string');
+    });
+
+    it('coerces numbers and booleans to strings', () => {
+      testUnknownAttributeAssignment(0, '0');
+      testUnknownAttributeAssignment(-1, '-1');
+      testUnknownAttributeAssignment(42, '42');
+      testUnknownAttributeAssignment(9000.99, '9000.99');
+      testUnknownAttributeAssignment(true, 'true');
+      testUnknownAttributeAssignment(false, 'false');
+    });
+
+    it('coerces NaN to strings and warns', () => {
+      spyOn(console, 'error');
+
+      testUnknownAttributeAssignment(NaN, 'NaN');
+      expectDev(normalizeCodeLocInfo(console.error.calls.argsFor(0)[0])).toBe(
+        'Warning: Received NaN for numeric attribute `unknown`. ' +
+          'If this is expected, cast the value to a string.\n' +
+          '    in div (at **)',
+      );
+      expectDev(console.error.calls.count()).toBe(1);
+    });
+
+    it('coerces objects to strings and does not warn', () => {
+      const lol = {
+        toString() {
+          return 'lol';
+        },
+      };
+
+      testUnknownAttributeAssignment({hello: 'world'}, '[object Object]');
```

Comment:
why do you want to set `[object Object]` instead of setting a property on the element? A custom element receiving `[object Object]` can't really do anything with it.

Reply:
So far, we’ve been focusing on the regular DOM elements so we didn’t change the behavior for custom elements. Given that 16 is imminent we’ll probably have to wait before 17 to change it.